test/dist_autograd_test.py

```diff
@@ -711,6 +713,21 @@ def test_backward_node_failure(self):
                 # Exit all other nodes.
                 pass
 
+    @dist_init(clean_shutdown=False)
+    def test_context_cleanup_timeout(self):
```

Comment:
This test should do the following:

1. Set the timeout to be 30s.
2. Rank 2 does an RPC to all other ranks.
3. Rank 2 tells all other ranks about its context id.
4. Kill rank 2 without cleaning up the context (shouldn't exit the `with` clause successfully)
5. All other ranks exit the `with` clause and then verify that rank 2's context id is cleaned up on their node.